Widgets/Widgets/OnThisDayView.swift

```diff
@@ -29,21 +29,26 @@ struct OnThisDayColors {
 
 struct OnThisDayView: View {
     @Environment(\.widgetFamily) private var widgetSize
+    @Environment(\.sizeCategory) var textSize
     var entry: OnThisDayProvider.Entry
 
+    var needsVerticalCompression: Bool {
+        return textSize == .extraExtraExtraLarge
+    }
+
     @ViewBuilder
     var body: some View {
         GeometryReader { proxy in
```

Comment:
Not worth messing with the delicate magical chemistry of SwiftUI for this now, but I think this `GeometryReader` is superfluous. I think you could get away with writing this kinda like:

```swift
@ViewBuilder
var body: some View {
	widgetContent
		.overlay(errorBox)
		.environment(\.layoutDirection, entry.isRTLLanguage ? .rightToLeft : .leftToRight)
		.widgetURL(entry.contentURL)
}

@ViewBuilder
var widgetContent: some View {
	switch widgetSize {
		...
	}
}

```

Reply:
100% in agreement. Good catch on this. I think there were a million readers at one point when I was trying to get the dot spacing working, and apparently I missed one in the removals.